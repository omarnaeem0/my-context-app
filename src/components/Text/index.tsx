import * as React from 'react';

export const Text = ({ children }: any) => (
  <div>{children}</div>
)