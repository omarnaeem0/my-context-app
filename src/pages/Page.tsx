import * as React from 'react';
import { MyConsumer } from '../context';


export const Page = () => (
  <MyConsumer>
    {(BB: any) => {
      // tslint:disable-next-line
      console.log(BB.Components)
      return(
        <BB.Components.Text>Hello</BB.Components.Text>
      )
    }}
  </MyConsumer>
)