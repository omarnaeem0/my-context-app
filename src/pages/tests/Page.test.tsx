import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import * as Component from '../../components';
import { Page } from '../Page';

beforeEach(() => {
  jest.resetModules();
});

const BB = {
  Components: {
    Text: Component.Text
  }
};
const mockContext = jest.fn();
jest.mock('../../context/consumer', () => ({
  MyConsumer: ({ children }: { children: any }) => children(BB)
}));

describe('Page', () => {
  beforeEach(() => {
    mockContext.mockReset();
  });

  test(`Should find the Page component`, () => {
    const component = TestRenderer.create(
      <Page />
    );
    const tree = component.toTree();
    expect(tree).toMatchSnapshot();
  })
});
