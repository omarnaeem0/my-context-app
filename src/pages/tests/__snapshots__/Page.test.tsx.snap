// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Should find the Page component 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {},
  "rendered": Object {
    "instance": null,
    "nodeType": "component",
    "props": Object {
      "children": [Function],
    },
    "rendered": Object {
      "instance": null,
      "nodeType": "component",
      "props": Object {
        "children": "Hello",
      },
      "rendered": Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": "Hello",
        },
        "rendered": Array [
          "Hello",
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  "type": [Function],
}
`;
